### PR TITLE
move report ID generation to server

### LINKED ISSUE #1346: Move reportID generation code to server.js
## Summary

move the reportID generator code to the server. The client will have to get the id on submission to display on the thank you page.

## Motivation

If the client generates the code and sends to the server, there is the potential for malicious actors to send fake reports with other people's report ID

### PR DESCRIPTION
**Pull request by sastels**
Fixes #1346 

# Description

Moves the code for generating the report ID to the server. This means that the client has to get the ID from the server to display on the thank you page.

To test:
1) get the COSMOSDB environment variables from me and add to your `f2/.env` file
2) `npm run build && node server.js`
3) go to localhost:3000, fill out the form, and submit
4) something like `Report NCFRS-r54dyetdphg saved to CosmosDB` will be printed to your console.
5) the same report ID will display on the thank you page.

# Checklist:

- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code

# Required followup work

If there's a problem saving to the database, this code will set the reportId to an error message such as "Error saving to CosmosDB". If that is the case (ie if the reportId doesn't start with "NCFRS", essentially) we should display a "please come back later" error message on the thank you page.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ data
 f2/build
 **/locales/*.js
 platform/**/.env*
+*.cer
```
Comment (sastels) [inline .gitignore]: stop any certs generated in testing from sneaking into the repo

**Comment by sastels** [comment]
Running the branch locally myself 
```
npm run build
node server.js
```
everything worked - in particular the last file in CosmosDB had the reportID that was displayed on the thankyou page. Is it possible that you weren't actually saving to the database (despite the server claiming that the file was saved?)

I can send you the COSMOSDB env vars that I'm using.

**Comment by ngosset** [comment]
> 
> 
> Running the branch locally myself
> 
> ```
> npm run build
> node server.js
> ```
> 
> everything worked - in particular the last file in CosmosDB had the reportID that was displayed on the thankyou page. Is it possible that you weren't actually saving to the database (despite the server claiming that the file was saved?)
> 
> I can send you the COSMOSDB env vars that I'm using.

I'll try again today.

I did review the record in CosmosDB so was definitely saving.

Maybe my browser was serving me a cached or old version of the react-app somehow, this would probably cause the issue, I'm sure I remembered to do the npm build but maybe I re-used an existing session

**Comment by ngosset** [comment]
Works for me now.

**Pull request merged**